lms/resources/_js_config/__init__.py

```diff
@@ -118,14 +118,29 @@ def enable_oauth2_redirect_error_mode(
 
         :param error_details: Technical details of the error
         :type error_details: str
-        :param auth_url: URL for the "Try again" button in the dialog
-        :type auth_url: str
+        :param auth_route: route for the "Try again" button in the dialog
+        :type auth_route: str
         :param is_scope_invalid: `True` if authorization failed because the
           OAuth client does not have access to all the necessary scopes
         :type is_scope_invalid: bool
         :param canvas_scopes: List of scopes that were requested
         :type canvas_scopes: list(str)
```

Comment:
Not sure which is the best way to go here, but I'm not sure there's inherently anything "Canvas"y about OAuth scopes (if that's what these are). So this could just be "required_scopes" or something.

On the other hand, if it's always Canvas why hide it? Maybe wait until there's something which isn't Canvas that needs this?

Reply:
In practice the only thing we use OAuth scopes for is for Canvas and what we do with them is Canvas-specific. For example they trigger a Canvas-specific error message about missing developer key scopes. I think if we one day had Blackboard scopes as well we might need a separate `blackboard_scopes` argument. Who knows